nilearn/experimental/plotting/_surface_plotting.py

```diff
@@ -1,36 +1,203 @@
 from __future__ import annotations
 
+from pathlib import Path
+
+import numpy
+
 from nilearn import plotting as old_plotting
-from nilearn.experimental.surface import SurfaceImage
+from nilearn._utils.docs import fill_doc
+from nilearn.experimental.surface import PolyMesh, SurfaceImage
+
+DEFAULT_HEMI = "left"
+
+# TODO double check types if we decide to keep them
+# SURF_MESH_TYPE = (
+#     str | Path | list[numpy.ndarray, numpy.ndarray] | Mesh | PolyMesh | None
+# )
+# MAP_TYPE = str | Path | numpy.ndarray | SurfaceImage | None
+
+
+def _check_inputs(
+    surf_map,
+    surf_mesh,
+    hemi: str,
+    bg_map=None,
+):
+    """Check inputs for surface plotting.
+
+    Where possible this will 'convert' the inputs to be able to pass them
+    to the the 'old' surface plotting functions.
+    """
+    if isinstance(surf_mesh, PolyMesh):
+        _check_hemi_present(surf_mesh, hemi)
+        surf_mesh = surf_mesh.parts[hemi]
+
+    if isinstance(surf_map, SurfaceImage):
+        if surf_mesh is None:
+            surf_mesh = surf_map.mesh.parts[hemi]
+        surf_map = surf_map.data.parts[hemi]
+
+    bg_map = _check_bg_map(bg_map, hemi)
+
+    return surf_map, surf_mesh, bg_map
+
+
+def _check_bg_map(bg_map, hemi: str) -> str | Path | numpy.ndarray | None:
+    """Return proper format of background map to be used."""
+    if isinstance(bg_map, SurfaceImage):
+        assert bg_map.data.parts[hemi] is not None
+        bg_map = bg_map.data.parts[hemi]
+    return bg_map
+
+
+def _check_hemi_present(mesh: PolyMesh, hemi: str) -> None:
+    """Check that a given hemisphere exists both in data and mesh."""
+    if hemi not in mesh.parts:
+        raise ValueError(f"{hemi} must be present in mesh")
 
```

Comment:
I wonder if some of the codecov complaints might actually be making sense this time.
Or is it that we only want to add tests for these once we make the surface stuff public?